core/services/headtracker/head_listener.go

```diff
@@ -156,19 +153,21 @@ func (hl *HeadListener) receiveHeaders(ctx context.Context, handleNewHead func(c
 
 // subscribe periodically attempts to connect to the ethereum node via websocket.
 // It returns true on success, and false if cut short by a done request and did not connect.
+// NOT THREAD SAFE
 func (hl *HeadListener) subscribe() bool {
-	hl.sleeper.Reset()
+	hl.subscribeRetryBackoff.Reset()
+
 	for {
 		if err := hl.unsubscribeFromHead(); err != nil {
```

Comment:
`unsubscribeFromHead` always returns `null`, maybe we shall change the function to return nothing and remove this `if`?

Reply:
Good catch! This code is very old and a bit wonky in places, will fix.